Source/Core/Gen8/Generators/WildGenerator8.cpp

```diff
@@ -18,26 +18,181 @@
+{
+    this->encounterArea = encounterArea;
+}
+
 std::vector<WildState> WildGenerator8::generate(u64 seed0, u64 seed1) const
 {
     Xorshift rng(seed0, seed1);
     rng.advance(initialAdvances + offset);
 
-    RNGList<u32, Xorshift, 2, 0> rngList(rng);
-
     std::vector<WildState> states;
-    for (u32 cnt = 0; cnt <= maxAdvances; cnt++, rngList.advanceState())
+    for (u32 cnt = 0; cnt <= maxAdvances; cnt++, rng.next())
     {
-        
+        WildState state(initialAdvances + cnt);
+        Xorshift gen(rng);
+        u8 slotPercent = gen.next<0, 100>();
+        gen.advance(84);
+        switch (encounter)
+        {
+        case Encounter::Grass:
+            state.setEncounterSlot(EncounterSlot::bdspSlot(slotPercent, encounter));
+            if (!filter.compareEncounterSlot(state))
+            {
+                continue;
+            }
+
+            state.setLevel(encounterArea.calcLevel(state.getEncounterSlot()));
+            break;
+        case Encounter::Surfing:
+        case Encounter::OldRod:
+        case Encounter::GoodRod:
+        case Encounter::SuperRod:
+            state.setEncounterSlot(EncounterSlot::bdspSlot(slotPercent, encounter));
+            if (!filter.compareEncounterSlot(state))
+            {
+                continue;
+            }
+
+            state.setLevel(encounterArea.calcLevel(state.getEncounterSlot(), gen.next<0, 10000>()));
+            break;
+        default:
+            break;
+        }
+
+        PersonalInfo info = PersonalLoader::getPersonal(Game::BDSP, encounterArea.getPokemon()[state.getEncounterSlot()].getSpecie(), 0);
+
+        gen.next(); // EC call
+        u32 sidtid = gen.next();
+        u32 pid = gen.next();
+
+        u16 fakeXor = (sidtid >> 16) ^ (sidtid & 0xffff) ^ (pid >> 16) ^ (pid & 0xffff);
+        u16 psv = ((pid >> 16) ^ (pid & 0xffff)) >> 4;
+        u16 realXor = (pid >> 16) ^ (pid & 0xffff) ^ tid ^ sid;
+
+        if (fakeXor < 16) // Force shiny
+        {
+            u8 shinyType = fakeXor == 0 ? 2 : 1;
+            state.setShiny(shinyType);
+            if (fakeXor != realXor)
+            {
+                u16 high = (pid & 0xFFFF) ^ tid ^ sid ^ (2 - shinyType);
+                pid = (high << 16) | (pid & 0xFFFF);
+            }
+        }
+        else // Force non shiny
+        {
+            state.setShiny(0);
+            if (psv == tsv)
+            {
+                pid ^= 0x10000000;
+            }
+        }
+
+        state.setPID(pid);
+
+        for (u8 i = 0; i < 6; i++)
+        {
+            state.setIV(i, gen.next() % 32);
+        }
+
+        state.setAbility(gen.next() % 2);
+
+        if (encounterArea.getLocation() > 222 && encounterArea.getLocation() < 244)
+        { // TODO: add unown check
+            u32 unownForm = gen.next() % 28; // Form call
+        }
+
+        if (lead == Lead::CuteCharm)
+        {
+            if (info.getGender() + 2 > 2)
```

Comment:
What exactly is this checking for?

```
if ((info.getGender() + 2 > 2) && (gen.next() % 3)
```

Will help with some of the ugliness of all these if else

Reply:
Should be cleaned up now